### PR TITLE
double event issue

### PR DESCRIPTION
**Pull request by Akhill2020**
**Description:** There was an issue in which if the event  is set for full day and more then one day then it will duplicate the event first day Also, in my testing, I have found that the event is not displayed on months last day so I have updated the event filter code and add some condition that prevent from duplicate event on the day.
**After:** https://www.screenpresso.com/=pPD51FQi7X9e
**clickup:** https://app.clickup.com/t/86cwkpk16
issue: https://wordpress.org/support/topic/all-day-events-are-duplicated-on-the-calendar/
https://wordpress.org/support/topic/deprecated-warning-25/
https://github.com/Xtendify/Simple-Calendar/issues/575

### CODE REVIEW

File: includes/calendars/views/default-calendar-grid.php [hunk 1]
```diff
@@ -363,22 +363,53 @@ private function draw_month($month, $year, $id = 0)
 			$timestamps = array_keys($events);
 			$lower_bound = array_filter($timestamps, [$this, 'filter_events_before']);
 			$higher_bound = array_filter($lower_bound, [$this, 'filter_events_after']);
-			$filtered =
-				is_array($events) && is_array($higher_bound) && !empty($events) && !empty($higher_bound)
-					? array_intersect_key($events, array_combine($higher_bound, $higher_bound))
-					: [];
 
-			// Put resulting events in an associative array, with day of the month as key for easy retrieval in calendar days loop.
+			$current_month = Carbon::createFromTimestamp($this->start, $calendar->timezone)->month;
```
Comment (rosinghal) [inline includes/calendars/views/default-calendar-grid.php:367]: I don't understand the changes starting from this line till 382. Can you please explain?

File: includes/calendars/views/default-calendar-grid.php [hunk 2]
```diff
@@ -363,22 +363,53 @@ private function draw_month($month, $year, $id = 0)
...
+				return false;
+			});
+
 			$day_events = [];
 			foreach ($filtered as $timestamp => $events_in_day) {
 				foreach ($events_in_day as $event) {
 					if ($event instanceof Event) {
-						$day = intval(Carbon::createFromTimestamp($timestamp, $calendar->timezone)->endOfDay()->day);
-						$day_events[$day][] = $event;
+						$event_start = Carbon::createFromTimestamp($event->start, $calendar->timezone);
```
Comment (rosinghal) [inline includes/calendars/views/default-calendar-grid.php:388]: Same here

File: includes/calendars/views/default-calendar-grid.php [hunk 3]
```diff
@@ -171,7 +171,7 @@ public function styles()
 	 *
 	 * @since  3.4.3
 	 */
-	public function format_timestamp($format = 'F', $timestamp)
+	public function format_timestamp($timestamp, $format = 'F')
```
[the remark this answers is not in the document]
Comment (Akhill2020) [reply includes/calendars/views/default-calendar-grid.php:174]: There was a deprecated warning in which it mentioned that we should not use option param on first position and mandatory on second.

File: includes/calendars/views/default-calendar-grid.php [hunk 4]
```diff
@@ -363,22 +363,44 @@ private function draw_month($month, $year, $id = 0)
...
-						$day_events[$day][] = $event;
+						$event_start = Carbon::createFromTimestamp($event->start, $calendar->timezone);
+						$event_end = Carbon::createFromTimestamp($event->end, $calendar->timezone);
+
+						for ($day = $event_start->copy(); $day->lte($event_end); $day->addDay()) {
+							if ($day->month == $current_month && $day->year == $current_year) {
+								$day_key = intval($day->format('j'));
+								$event_id = $event->uid;
+
+								if ($day->isSameDay($event_end) && $event_end->hour == 0 && $event_end->minute == 0) {
```
[the remark this answers is not in the document]
Comment (Akhill2020) [reply includes/calendars/views/default-calendar-grid.php:389]: Done

**Pull request merged**